src/atc/schema_manager/schema_manager.py

```diff
@@ -0,0 +1,107 @@
+import copy
+from typing import Dict
+
+import pyspark.sql.types as T
+
+from atc.atc_exceptions import AtcException
+from atc.config_master import TableConfigurator
+from atc.singleton import Singleton
+
+
+class NoSuchSchemaException(AtcException):
+    pass
+
+
+class FalseSchemaDefinitionException(AtcException):
+    pass
+
+
+class UnregisteredSchemaDefinitionException(AtcException):
+    pass
+
+
+class SchemaManager(metaclass=Singleton):
+
+    # This dict contains the registered schemas
+    _registered_schemas: Dict[str, T.StructType] = {}
+
+    def __init__(self):
+        pass
+
+    def register_schema(self, schema_name: str, schema: T.StructType):
+        self._registered_schemas[schema_name] = schema
+
+    def get_schema(self, schema_identifier: str):
+        # Check if the identifier is a registered schema
+        if schema_identifier in self._registered_schemas.keys():
+            return self._registered_schemas[schema_identifier]
+
+        # Otherwise, check if the schema identifier is a table identifier
+        try:
+            schema = TableConfigurator().table_property(
+                table_id=schema_identifier, property_name="schema"
+            )
+        except ValueError:
+            raise NoSuchSchemaException(schema_identifier)
+
+        # If the schema is a string, check if it has been registered
+        if isinstance(schema, str) and schema in self._registered_schemas.keys():
+            return self._registered_schemas[schema]
```

Comment:
insert the schema into the registry before you return it. That way the next call is faster